WhoIsThat/WhoIsThat/Connections/RestService.cs

```diff
@@ -15,82 +16,118 @@ namespace WhoIsThat.Connections
             }
 
             catch (Exception exception)
             {
-                throw exception;
+                //After using this API call in every code part should be checked if it's null and FatalStorageError message should be displayed
+                return null;
             }
         }
 
+        /// <inheritdoc/>
         public async Task<ImageObject> CreateImageObject(ImageObject personObject)
         {
-            try
+            const string restUrl = "https://teststorageserver.azurewebsites.net/api/images/add";
+            var uri = new Uri(string.Format(restUrl, string.Empty));
+
+            var jsonContent = JsonConvert.SerializeObject(personObject, new JsonSerializerSettings
             {
-                string restUrl = "https://teststorageserver.azurewebsites.net/api/images/add";
-                var uri = new Uri(string.Format(restUrl, string.Empty));
+                NullValueHandling = NullValueHandling.Ignore
+            });
+
+            var request = new HttpRequestMessage(HttpMethod.Post, uri)
+            {
+                Content = new StringContent(jsonContent, Encoding.UTF8, "application/json")
+            };
+
+            var response = (await HttpClient.SendAsync(request)).EnsureSuccessStatusCode();
 
-                var response = await _httpHandler.Post(uri, personObject);
-                if (response.IsSuccessStatusCode)
-                {
-                    var content = await response.Content.ReadAsStringAsync();
-                    return JsonConvert.DeserializeObject<ImageObject>(content);
-                }
-                else
-                {
-                    throw new Exception("Something went wrong: " + response.StatusCode);
-                }
+            var responseContent = await response.Content.ReadAsStringAsync();
+            try
+            {
+                var responseObject = JsonConvert.DeserializeObject<ImageObject>(responseContent);
+                return responseObject;
             }
-            catch (Exception exception)
+
+            catch (JsonException jsonException)
             {
-                throw exception;
+                //Mapped error message needs to be displayed after checking
+                return null;
             }
         }
+
+        /// <inheritdoc/>
         public async Task<string> Identify()
         {
-            string restUrl = "https://testrecognition.azurewebsites.net/api/recognitionservices/identify";
+            const string restUrl = "https://testrecognition.azurewebsites.net/api/recognitionservices/identify";
             var uri = new Uri(string.Format(restUrl, string.Empty));
 
-            var response = await _httpHandler.Get(uri);
-            if (response.IsSuccessStatusCode)
+            var response = await HttpClient.GetAsync(uri);
+            if (!response.IsSuccessStatusCode) return Constants.FatalRecognitionError;
```

Comment:
Throw, dont return

Reply:
Fair point, I need exception class in frontend too, thanks for reminding me!